src/test/integration/tests/basic.ts

```diff
@@ -0,0 +1,52 @@
+/**
+ * @license
+ * Copyright (c) 2020 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at
+ * http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at
+ * http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+import {html} from 'lit-html';
+import { SSRTest } from './ssr-test';
+
+export const tests = {
+  'textExpression': {
+    render(x: any) {
+      return html`<div>${x}</div>`;
+    },
+    expectations: [
+      {
+        args: ['TEST'],
+        html: '<div>TEST</div>'
```

Comment:
I mentioned this IRL the other day, but seems likely that we'll want more than just semantic DOM diffing in the future (e.g. for testing node identity with logic too complex for `stableSelectors`, etc.).  I did a little local hacking to add an optional `verify(container)` field to the `SSRTest:expectations` type that can run arbitrary assertions on the passed DOM, and the only requirement is that whatever dependencies that method needs is either importable on the server (doesn't need to be runnable, since `verify()` would only be called on the client), or else dynamically imported.  It seems workable so I think we can cross that bridge and add something like that when the need arises.

Curiously though, just loading `@open-wc/testing` in this file seems to mess up the tests; I didn't debug if that was due to something messing up on the server or client though...

Reply:
Ack.

Another thing we can try out is to find a way to associate a test with a client-side module that adds the additional checks. Right now basic_test.ts imports this and runs one generic test loop. It could potentially abstract that test runner and let a non-generic client test pick out one of the integration tests, then do additional interactions/assertions.